nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/SdkPackageUpdater.cs

```diff
@@ -254,62 +254,133 @@ private static UpdateResult TryUpdateDependencyVersion(ImmutableArray<ProjectBui
+                            logger.Log($"    Found correct [{packageNode.Name}] version attribute in [{buildFile.RepoRelativePath}].");
+                            foundCorrect = true;
+                        }
+                    }
                 }
-                // Is this the case that the version is specified directly in the package node?
-                else
+                else if (versionElement is not null)
                 {
-                    var currentVersion = versionAttribute.Value.TrimStart('[', '(').TrimEnd(']', ')');
-                    if (currentVersion.Contains(',') || currentVersion.Contains('*'))
-                    {
-                        logger.Log($"    Found unsupported [{packageNode.Name}] version attribute value [{versionAttribute.Value}] in [{buildFile.RepoRelativePath}].");
-                        foundUnsupported = true;
-                    }
-                    else if (currentVersion == previousDependencyVersion)
+                    var versionValue = versionElement.GetContentValue();
+                    if (versionValue.StartsWith("$(") && versionValue.EndsWith(")"))
                     {
-                        logger.Log($"    Found incorrect [{packageNode.Name}] version attribute in [{buildFile.RepoRelativePath}].");
-                        updateAttributes.Add(versionAttribute);
+                        propertyNames.Add(versionValue.Substring(2, versionValue.Length - 3));
                     }
-                    else if (previousDependencyVersion == null && NuGetVersion.TryParse(currentVersion, out var previousVersion))
+                    else
                     {
-                        var newVersion = NuGetVersion.Parse(newDependencyVersion);
-                        if (previousVersion < newVersion)
+                        var currentVersion = versionValue.TrimStart('[', '(').TrimEnd(']', ')');
+                        if (currentVersion.Contains(',') || currentVersion.Contains('*'))
+                        {
+                            logger.Log($"    Found unsupported [{packageNode.Name}] version node value [{versionValue}] in [{buildFile.RepoRelativePath}].");
+                            foundUnsupported = true;
+                        }
+                        else if (currentVersion == previousDependencyVersion)
                         {
-                            previousPackageVersion = currentVersion;
+                            logger.Log($"    Found incorrect [{packageNode.Name}] version node in [{buildFile.RepoRelativePath}].");
+                            if (versionElement is XmlElementSyntax elementSyntax)
+                            {
+                                updateNodes.Add(elementSyntax);
+                            }
+                            else
+                            {
+                                throw new InvalidDataException("A concrete type was required for updateNodes. This should not happen.");
```

Comment:
Can you provide an example of when we could hit this exception?

Reply:
Adding a comment: `// This only exists for completeness in case we ever add a new type of node we don't want to silently ignore them.`